modules/systemd.nix

```diff
@@ -286,71 +294,73 @@ in {
   # If we run under a Linux system we assume that systemd is
   # available, in particular we assume that systemctl is in PATH.
   # Do not install any user services if username is root.
-  config = mkIf (pkgs.stdenv.isLinux && config.home.username != "root") {
-    xdg.configFile = mkMerge [
-      (lib.listToAttrs ((buildServices "service" cfg.services)
-        ++ (buildServices "slice" cfg.slices)
-        ++ (buildServices "socket" cfg.sockets)
-        ++ (buildServices "target" cfg.targets)
-        ++ (buildServices "timer" cfg.timers)
-        ++ (buildServices "path" cfg.paths)
-        ++ (buildServices "mount" cfg.mounts)
-        ++ (buildServices "automount" cfg.automounts)))
-
-      sessionVariables
-
-      settings
-    ];
-
-    # Run systemd service reload if user is logged in. If we're
-    # running this from the NixOS module then XDG_RUNTIME_DIR is not
-    # set and systemd commands will fail. We'll therefore have to
-    # set it ourselves in that case.
-    home.activation.reloadSystemd = hm.dag.entryAfter [ "linkGeneration" ] (let
-      cmd = {
-        suggest = ''
-          PATH=${dirOf cfg.systemctlPath}:$PATH \
-          bash ${./systemd-activate.sh} "''${oldGenPath=}" "$newGenPath"
-        '';
-        legacy = ''
-          PATH=${dirOf cfg.systemctlPath}:$PATH \
-          ${pkgs.ruby}/bin/ruby ${./systemd-activate.rb} \
-            "''${oldGenPath=}" "$newGenPath" "${servicesStartTimeoutMs}"
-        '';
-        sd-switch = let
-          timeoutArg = if cfg.servicesStartTimeoutMs != 0 then
-            "--timeout " + servicesStartTimeoutMs
-          else
-            "";
-        in ''
-          ${pkgs.sd-switch}/bin/sd-switch \
-            ''${DRY_RUN:+--dry-run} $VERBOSE_ARG ${timeoutArg} \
-            ''${oldGenPath:+--old-units $oldGenPath/home-files/.config/systemd/user} \
-            --new-units $newGenPath/home-files/.config/systemd/user
-        '';
-      };
+  config = mkIf (cfg.enable == true && pkgs.stdenv.isLinux
```

Comment:
`pkgs.stdenv.isLinux` should probably be moved to an assertion

Reply:
What do you mean by that, do you have an example?

Reply:
I did something, let me know if it's OK

Reply:
Yes, that's good